network/src/commonTest/kotlin/com/wire/kalium/api/tools/json/api/asset/AssetApiTest.kt

```diff
@@ -0,0 +1,162 @@
+import com.wire.kalium.network.api.asset.AssetApiImp
+import com.wire.kalium.network.api.model.AssetMetadata
+import com.wire.kalium.network.api.model.AssetRetentionType
+import com.wire.kalium.network.exceptions.KaliumException
+import com.wire.kalium.network.utils.NetworkResponse
+import com.wire.kalium.network.utils.isSuccessful
+import io.ktor.http.HttpStatusCode
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.test.runTest
+import kotlin.test.Test
+import kotlin.test.assertEquals
+import kotlin.test.assertTrue
+
+@ExperimentalCoroutinesApi
+class AssetApiTest : ApiTest {
+    @Test
+    fun givenAValidAssetUploadApiRequest_whenCallingTheAssetUploadApiEndpoint_theRequestShouldBeConfiguredCorrectly() = runTest {
+        // Given
+        val assetMetadata = AssetMetadata("image/jpeg", true, AssetRetentionType.ETERNAL, "md5-hash")
+        val encryptedData = ByteArray(16)
+        val httpClient = mockAuthenticatedHttpClient(
+            VALID_ASSET_UPLOAD_RESPONSE.rawJson,
+            statusCode = HttpStatusCode.Created,
+            assertion = {
+                assertPost()
+                assertJson()
+                assertNoQueryParams()
+                assertAuthorizationHeaderExist()
+                assertPathEqual(PATH_PUBLIC_ASSETS)
+            }
+        )
+
+        // When
+        val assetApi: AssetApi = AssetApiImp(httpClient)
+        val response = assetApi.uploadAsset(assetMetadata, encryptedData)
+
+        // Then
+        assertTrue(response.isSuccessful())
+        assertEquals(response.value, VALID_ASSET_UPLOAD_RESPONSE.serializableData)
+    }
+
+    @Test
+    fun givenAValidAssetUploadApiRequest_whenCallingTheAssetUploadApiEndpoint_theRequestHeaderShouldContainTheAssetToken() = runTest {
+        // Given
+        val assetMetadata = AssetMetadata("image/jpeg", true, AssetRetentionType.ETERNAL, "md5-hash")
+        val encryptedData = ByteArray(16)
+        val httpClient = mockAuthenticatedHttpClient(
+            VALID_ASSET_UPLOAD_RESPONSE.rawJson,
+            statusCode = HttpStatusCode.Created,
+            assertion = {
+                assertPost()
+                assertJson()
+                assertNoQueryParams()
+                assertAuthorizationHeaderExist()
+                assertPathEqual(PATH_PUBLIC_ASSETS)
+            }
+        )
+
+        // When
+        val assetApi: AssetApi = AssetApiImp(httpClient)
+        val response = assetApi.uploadAsset(assetMetadata, encryptedData)
+
+        // Then
+        assertTrue(response.isSuccessful())
+        assertEquals(response.value, VALID_ASSET_UPLOAD_RESPONSE.serializableData)
+    }
+
+    @Test
+    fun givenAnInvalidAssetUploadApiRequest_whenCallingTheAssetUploadApiEndpoint_theRequestShouldContainAnError() = runTest {
+        // Given
+        val assetMetadata = AssetMetadata("image/jpeg", true, AssetRetentionType.ETERNAL, "md5-hash")
+        val encryptedData = ByteArray(16)
```

Comment:
```suggestion
        val encryptedData = ByteArray(16)
        Random.nextBytes(encryptedData)
```
we can add this line here just to generate some random data